### ISSUE TITLE
Download table data into csv - Commas screw all the headers up

### ISSUE THREAD

**Issue by ARWattam** [PATRIC3/patric3_website#2352]
I have noticed this myself, but this if from a PATRIC user describing his problems with downloading tables with csv format.
"For example, upon export to a .csv, “Oligopeptide ABC transporter, ATP-binding protein OppF (TC 3.A.1.5.1)” not surprisingly gets split into two cells: “Oligopeptide ABC transporter” and “ ATP-binding protein OppF (TC 3.A.1.5.1)”.  Note the space at the beginning of the second entry that basically confirms this is the source of the odd behavior.  Not sure how you would want to handle this.  Possibly replacing all the description commas with semicolons or other characters prior to CSV export?"

It is true that this is a REAL PAIN, especially when you are downloading a huge table.

**Comment by dmachi** [comment]
Download it as a TSV (tab separated values) which is just a CSV with tabs separating fields. I believe we quote the CSV elements, but if the program doing the reading of the csv file doesn’t handle that,  then it will have behavior like this.

**Comment by ARWattam** [comment]
Excellent point.  I have passed that on to the user and he was somewhat embarrassed.  He hadn't made the connection that it was a tsv file, but didn't know exactly what he thought it would be.  I think we can consider the issue closed as it met his requirements....unless we want to consider putting quotation marks around protein family names in the csv so we don't have to address this again.

**Comment by nconrad** [comment]
No, that file was a csv file.  I haven't looked at the data api code yet, but `Oligopeptide ABC transporter, ATP-binding protein OppF (TC 3.A.1.5.1)` is appearing without quotes in the header of the csv file.  I think this is bug that needs to be fixed.

**Comment by dmachi** [comment]
It wasn’t a TSV file, so he didn’t do anything wrong in that regard.  The data api lets you download CSV or TSV files, so I just meant TSV was a work around.  I haven’t checked our code to see whether the CSV error is on our side or on his.  I thought that we were already quoting the individual fields in our CSV output (though I could be wrong) and so I was saying that the problem could be on the user’s parsing side if he is using a simple split on commas approach.
